### ISSUE TITLE
Open-Vocabulary semantic segmentation

### ISSUE THREAD

**Issue by z2587721154** [google-research/big_vision#171]
What parameters are fine-tuned for siglip2 using catseg's line for  Open-Vocabulary semantic segmentation in this paper?

**Comment by ferjad** [comment]
I ll have to check the code to see if we mapped from 768 to 512 but the goal is to stick to cat-seg as closely as possible so learning rate and all other hyper parameters including which layers are unfrozen are exactly the same. If the projection is an important detail for you, I can check it next week. The tokenizer for open vocabulary stage needs to match the one used for contrastive training so we use the same one as siglip2 for the experiments.

**Comment by z2587721154** [comment]
> 我必须检查代码，看看我们是否从 768 映射到 512，但目标是尽可能坚持 cat-seg，因此学习率和所有其他超参数（包括哪些层被解冻）完全相同。如果预测对你来说是一个重要的细节，我可以在下周检查一下。开放词汇阶段的分词器需要与用于对比训练的分词器匹配，因此我们在实验中使用与 siglip2 相同的分词器。

Thank you very much for your help! This project is very important to me. Currently, my open-vocabulary semantic segmentation model, built on a similar CatSeg model using CLIP, has achieved new SOTA results. However, when reproducing the CatSeg model using Siglip2, I noticed that the early checkpoint performance was suboptimal, and the final results were not as expected. I have been trying to find the root cause but have not yet found a clear solution.

If you could share some implementation details of Siglip2 or key code snippets from your use of Siglip2 with CatSeg, I would greatly appreciate it. I firmly believe that Siglip2 will provide a significant boost to my model, helping me achieve better performance.

Thank you again for your support! I look forward to your help. My email is: 2587721154@qq.com.

**Comment by ChaojunZ** [comment]
> I ll have to check the code to see if we mapped from 768 to 512 but the goal is to stick to cat-seg as closely as possible so learning rate and all other hyper parameters including which layers are unfrozen are exactly the same. If the projection is an important detail for you, I can check it next week. The tokenizer for open vocabulary stage needs to match the one used for contrastive training so we use the same one as siglip2 for the experiments.

I have the same problem, how to solve this.

**Comment by ferjad** [comment]
Did you folks change the image processing to be compatible with big vision checkpoints? Namely we do not normalize the image with mean and std but rather between -1 and 1.

**Comment by z2587721154** [comment]
> Did you folks change the image processing to be compatible with big vision checkpoints? Namely we do not normalize the image with mean and std but rather between -1 and 1.

Thank you very much for your help, I found out that I was using 

```
clip_images = [(x - self.clip_pixel_mean) / self.clip_pixel_std for x in images]

clip_images = ImageList.from_tensors(clip_images, self.size_divisibility)

clip_images_resized = F.interpolate(clip_images.tensor, size=self.clip_resolution, mode='bilinear', align_corners=False, )
```
 I'll further replace it with 
```
pp_img = pp_builder.get_preprocess_fn( f'resize({RES})|value_range(-1, 1)')
imgs = np.array([pp_img({'image': np.array(image)})['image'] for image in images])
```
and the clip used by catseg turns the number of channels 768 into 512 channels in the last layer through dense processing, have you been dense? The second question is: Does the loss function of catseg need to be modified? Do you have any plans to publish or share catseg_v1 code that uses siglip2?
